specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,988 @@
+is considered as supporting Adaptive Profile programs. An adaptive profile must indicate what 
+additional capabilities it uses via module flags.
+Ideally, tools should be able
+to indicate what capabilities of the Adaptive Profile a back-end is implementing
+and should run a verification pass to ensure capabilities of the Adaptive Profile
+a back-end are not implementing are properly rejected.
+More details about each of the capabilities
+are outlined below.
+
+## Mandatory Capabilities
+
+**Bullet 1: Quantum transformations** <br/>
+
+The set of available instructions that transform the quantum state may vary
+depending on the targeted backend. The profile specification defines how to
+leverage and combine the available instructions to express a program, but does
+not dictate which quantum instructions may be used. Targeting a program to a
+specific backend requires choosing a suitable profile and quantum instruction
+set (QIS). Both can be chosen largely independently, though certain instruction
+sets may be incompatible with this (or other) profile(s). The section on the
+[quantum instruction set](#quantum-instruction-set) defines the requirements for
+a QIS to be compatible with the Adaptive Profile. 
+More information about the role of
+the QIS, recommendations for front- and backend providers, as well as the
+distinction between run-time functions and quantum instructions can be found in
+[this document](../Instruction_Set.md).
+
+**Bullet 2: Measurements** <br/>
+
+The second requirement relieves a restriction from the base profile that qubits
+can only be measured at the end of the program and that no quantum operations
+can be performed on a measured qubit. 
+The only requirement here is that all qubits must be measurable, but
+there are no restrictions on when these measurements are made, and there is
+no restriction necessitating that all qubits be measured. I.E. it's perfectly
+valid to measure a subset of qubits used in the program, not have all measured
+qubits in the output, and perform the measurements at different points.
+
+**Bullet 3: Program output** <br />
+
+There is no change to output specifications with respect to the base profile.
+
+The specification of QIR and all its profiles needs to permit to accurately
+reflect the program intent. This includes being able to define and customize the
+program output. The Base Profile specification hence requires explicitly
+expressing which values/measurements are returned by the program and in which
+order. How to express this is defined in the section on [output
+recording](#output-recording).
+
+While it is sufficient for the QPU to do a final measurement of all qubits in a
+predefined order at the end of the program, only the selected subset will be
+reflected in the produced output schema. A suitable output schema can be 
+generated in a post-processing step after the computation on the quantum
```

Comment:
Seems like a discussion topic for both this and the base profile spec.

Reply:
We can rephrase this both here and in the Base Profile to say something along the lines of "independently of the quantum execution" rather than after.